### PR TITLE
Fix Geothermal parsing in TAIPOWER

### PR DESCRIPTION
**Pull request by silimotion**
## Issue

The recent #5930 fixed and updated the parser but ignored geothermal values.

## Description

This PR adds the geothermal values. It changes the API endpoint to the english version to avoid having to search for chinese strings, which would have been needed to get the goethermal values using the previous endpoint. 

### Preview
```
Parser result:
[{'capacity': {'biomass': 38.2,
               'coal': 14697.2,
               'gas': 18060.6,
               'geothermal': 7.2,
               'hydro': 2101.9999999999995,
               'nuclear': 1902.0,
               'oil': 1592.5,
               'solar': 10822.4,
               'unknown': 626.9,
               'wind': 958.0},
  'correctedModes': [],
  'datetime': datetime.datetime(2023, 9, 24, 2, 30, tzinfo=<DstTzInfo 'Asia/Taipei' CST+8:00:00 STD>),
  'production': {'biomass': 27.0,
                 'coal': 12303.4,
                 'gas': 12093.7,
                 'geothermal': 2.8,
                 'hydro': 453.7,
                 'nuclear': 1879.6,
                 'oil': 350.7,
                 'solar': 0.0,
                 'unknown': 625.3,
                 'wind': 51.0},
  'source': 'taipower.com.tw',
  'sourceType': <EventSourceType.measured: 'measured'>,
  'storage': {'hydro': 785.7},
  'zoneKey': 'TW'}]
---------------------
took 1.60s
min returned datetime: 2023-09-23 18:30:00+00:00 UTC
max returned datetime: 2023-09-23 18:30:00+00:00 UTC  -- OK, <2h from now :) (now=2023-09-23T18:34:28+00:00 UTC)
```
### Double check

- [X] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: parsers/TAIPOWER.py [hunk 1]
```diff
@@ -94,18 +96,18 @@ def fetch_production(
...
-        "nuclear": ["Nuclear"],
-        "solar": ["Solar"],
-        "wind": ["Wind"],
-        "unknown": ["Co-Gen"],
+        "oil": ["OIL", "DIESEL"],
+        "hydro": ["HYDRO"],
+        "nuclear": ["NUCLEAR"],
+        "solar": ["SOLAR"],
+        "wind": ["WIND"],
+        "unknown": ["COGEN", ""],
```
Comment (VIKTORVAV99) [inline parsers/TAIPOWER.py:107]: any specific reason `""` was included?

Comment (silimotion) [reply parsers/TAIPOWER.py]: Must have been a typo

**Review by VIKTORVAV99 — COMMENTED** [review]
As far as I can tell this looks good but there is a small issue in the return type for fetch_production where it should be `list[dict]` instead of `dict`.

Would be nice to get that included in this PR as well :)

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

**Pull request merged**